### ISSUE TITLE
Virtual text + relativenumber rendering issues

### ISSUE THREAD

**Issue by lukas-reineke** [neovim/neovim#17638]
### Neovim version (nvim -v)

NVIM v0.7.0-dev+1203-gc3cc17f0e

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Arch linux

### Terminal name/version

Kitty 0.24.4

### $TERM environment variable

xterm-kitty

### Installation

build from repo

### How to reproduce the issue

 `nvim -u init.lua`

```lua
-- init.lua
vim.wo.number = true
vim.wo.relativenumber = true

vim.cmd [[normal 20ohello]]
vim.cmd [[normal ithis is weird]]

vim.api.nvim_buf_set_extmark(
    0,
    vim.api.nvim_create_namespace("foo"),
    1,
    0,
    {virt_text = {{"hello", "String"}}, virt_text_pos = "overlay", hl_mode = "combine", virt_text_win_col = 20, priority = 1}
)
```

### Expected behavior

Virtual text is displayed normally.

### Actual behavior

The line with the virtual text has render artifacts from the line the cursor is on.

https://user-images.githubusercontent.com/12900252/157021012-f829d574-f2b3-427d-ae0e-3ae42bfe8a20.mp4

**Comment by lukas-reineke** [comment]
\cc @bfredl

**Comment by zeertzjq** [comment]
Looks like a very recent regression.

**Comment by clason** [comment]
@lukas-reineke can you bisect?

**Comment by clason** [comment]
Yeah, seems to have been introduced by #16897.

**Comment by lewis6991** [comment]
Yep, specifically this line which was moved outside the below `if`: https://github.com/neovim/neovim/blob/c3cc17f0e61d70436c601444298fceeafbfbacb3/src/nvim/screen.c#L2211

**Comment by lewis6991** [comment]
Dirty fix: #17640
